Tasks/Gradle/CodeAnalysis/Common/CodeAnalysisResultPublisher.ts

```diff
@@ -24,8 +24,20 @@ export class CodeAnalysisResultPublisher {
      *
      * @param {string} prefix - used to discriminate between artifacts comming from different builds of the same projects (e.g. the build number)
      */
-    public uploadArtifacts(prefix: string):void {
+    public uploadArtifacts(prefix: string):number {
+        // If there are no results to upload, return
         if (this.analysisResults.length === 0) {
+            tl.debug("Skipping artifact upload: No analysis results");
+            return;
+        }
+
+        // If there are no files to upload, return
+        var analysisResultsWithFiles:AnalysisResult[] = this.analysisResults.filter(
+            (analysisResult:AnalysisResult) => {
+                return analysisResult.resultFiles != undefined && analysisResult.resultFiles != null && analysisResult.resultFiles.length > 0;
+        });
+        if (analysisResultsWithFiles.length === 0) {
```

Comment:
This is the core of the fix - if no analysisResult objects had files to upload, then return early.